File: 본문/python/Code_3_04_2.py
# 입력
N = int(input())
P = [ None ] * N
Q = [ None ] * N
for i in range(N):
	P[i], Q[i] = map(int, input().split())

# 답 계산
answer = 0
for i in range(N):
	answer += Q[i] / P[i]

# 출력
print("{:.12f}".format(answer))

File: 본문/java/Code_3_04_2.java
import java.util.*;

class Code_3_04_2 {
	public static void main(String[] args) {
		// 입력
		Scanner sc = new Scanner(System.in);
		int N = sc.nextInt();
		int[] P = new int[N + 1];
		int[] Q = new int[N + 1];
		for (int i = 1; i <= N; i++) {
			P[i] = sc.nextInt();
			Q[i] = sc.nextInt();
		}
		
		// 답 계산/출력
		double answer = 0.0;
		for (int i = 1; i <= N; i++) {
			answer += (double)Q[i] / P[i];
		}
		System.out.format("%.12f\n", answer);
	}
}
